test/models/partner_test.rb

```diff
@@ -14,44 +14,60 @@ class PartnerTest < ActiveSupport::TestCase
     assert @user.partner_admin?
   end
 
-  test 'fields validate' do
+  test 'validate title length' do
     partner = Partner.new
     partner.save
 
     # The only mandatory field is title (for now)
-    assert_equal ["can't be blank", "must be at least 5 characters long"], partner.errors[:name], "must have a name"
-    assert_equal 1, partner.errors.keys.length
+    assert error_map = partner.errors.where(:name, :too_short).first
+    assert_equal 'must be at least 5 characters long', error_map.options[:message]
+    assert_equal 1, partner.errors.attribute_names.length
+  end
```

Comment:
Error maps are now error objects and should be treated like that, this fixes the error created by that, apparently the rest of the code doesn't trigger a warning or an error so I figure it's fine.

Reply:
Also, I split these up into separate tests rather than one big monstrous test haha